service/src/test/java/uk/nhs/adaptors/gp2gp/TransformJsonToXml.java

```diff
@@ -0,0 +1,209 @@
+import uk.nhs.adaptors.gp2gp.ehr.mapper.OrganizationToAgentMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.OutputMessageWrapperMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ParticipantMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.
+        PertinentInformationObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.
+        PractitionerAgentPersonMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.RequestStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.StructuredObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.DiagnosticReportMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.ObservationMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.SpecimenMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.EhrExtractTemplateParameters;
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.MessageContext;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TransformJsonToXml {
+
+    private static final String JSON_FILE_INPUT_PATH =
+            Paths.get("src", "test", "resources", "JsonToXml", "Input").toFile().getAbsoluteFile().getAbsolutePath() + "/";
+    private static final String XML_OUTPUT_PATH =
+            Paths.get("src", "test", "resources", "JsonToXml", "Output").toFile().getAbsoluteFile().getAbsolutePath() + "/";
+
+    public static void main(String[] args) throws Exception {
+
+        var inputWrapper = getFiles();
+
+        for (int i = 0; i < inputWrapper.getJsonFileInputs().size(); i++) {
+            var jsonString = inputWrapper.getJsonFileInputs().get(i);
+            String xmlResult = mapJsonToXml(jsonString);
+            var fileName = inputWrapper.getJsonFileNames().get(i);
+            writeToFile(xmlResult, fileName);
+        }
+    }
+
+    private static InputWrapper getFiles() {
+
+        File[] files = new File(JSON_FILE_INPUT_PATH).listFiles();
+        List<String> jsonStringInputs = new ArrayList<>();
+        List<String> fileNames = new ArrayList<>();
+
+        for (File jsonFile : files) {
+            if (FilenameUtils.getExtension(jsonFile.getName()).equalsIgnoreCase("json")) {
+                try {
+                    String jsonAsString = readJsonFileAsString(JSON_FILE_INPUT_PATH + jsonFile.getName());
+                    jsonStringInputs.add(jsonAsString);
+                    fileNames.add(jsonFile.getName());
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+        return InputWrapper.builder()
+                .jsonFileInputs(jsonStringInputs)
+                .jsonFileNames(fileNames)
+                .build();
+    }
+
+    private static String stringToJson(String json) throws JSONException {
```

Comment:
I am not sure we can assume patient resource or nhs identifier will be first in the list always. It would be good to parse the resource to fhir bundle and filter patient resource and valid identifier

something like that, with usage of FhirParseService we have already, you can search for it in other tests.
```
 Bundle bundle = new FhirParseService().parseResource(json, Bundle.class);
        return bundle.getEntry().stream()
            .map(Bundle.BundleEntryComponent::getResource)
            .filter(resource -> ResourceType.Patient.equals(resource.getResourceType()))
            .map(resource -> (Patient)resource)
            .map(resource -> getNhsNumberIdentifier(nhsnumberSystem, resource))
            .filter(Optional::isPresent)
            .map(identifier -> identifier.get().getValue())
            .findFirst()
            .orElseThrow(() -> new InvalidJsonException("No nhs number in Patient resource"));
```
The method could be called extractNhsNumber or getNhsNumber to reflect what it is doing.